### ISSUE TITLE
Loss is nan, stopping training

### ISSUE THREAD

**Issue by AhmedHashish123** [microsoft/unilm#551]
**Describe**
Model I am using (BEiT):
I tried pretraining on a custom dataset. I used some dummy data which contains 5 classes and 12 images for each class. The dataset structure is like this:

`
train/
    class1/
      img1.jpg
    class2/
      img2.jpg
`

So, "train" directory contains several subdirectories called "class1, class2, class3...". In each subdirectory there are 12 images.

- I get this error in the first epcoh in training: Loss is nan, stopping training
- The images' resolution is 224.
- The script used:
!python run_beit_pretraining.py --data_path /content/unilm/beit/dataset_for_pre_training/train/ --output_dir /content/unilm/beit/pretrained_model_directory/ --num_mask_patches 1 --model beit_base_patch16_224_8k_vocab --discrete_vae_weight_path /content/unilm/beit/dall_e_tokenizer_weight/ --batch_size 2 --lr 1.5e-3 --warmup_steps 10 --epochs 5 --clip_grad 3.0 --drop_path 0 --layer_scale_init_value 0.1

Any idea why this error occurs? Thanks in advance.

**Comment by AhmedHashish123** [comment]
I changed --num_mask_patches 1 to --num_mask_patches 75 and now the training is working. I don't quite understand what this parameter does. Can someone explain?

**Comment by donglixp** [comment]
The arg `--num_mask_patches` indicates how many patches we would like to mask. For example, `--num_mask_patches 75` masks ~40% (=75/196) of total patches.

**Comment by umer-newgrad** [comment]
Did the pretraining work for you? 

> **Describe** Model I am using (BEiT): I tried pretraining on a custom dataset. I used some dummy data which contains 5 classes and 12 images for each class. The dataset structure is like this:
> 
> `train/ class1/ img1.jpg class2/ img2.jpg`
> 
> So, "train" directory contains several subdirectories called "class1, class2, class3...". In each subdirectory there are 12 images.
> 
> * I get this error in the first epcoh in training: Loss is nan, stopping training
> * The images' resolution is 224.
> * The script used:
>   !python run_beit_pretraining.py --data_path /content/unilm/beit/dataset_for_pre_training/train/ --output_dir /content/unilm/beit/pretrained_model_directory/ --num_mask_patches 1 --model beit_base_patch16_224_8k_vocab --discrete_vae_weight_path /content/unilm/beit/dall_e_tokenizer_weight/ --batch_size 2 --lr 1.5e-3 --warmup_steps 10 --epochs 5 --clip_grad 3.0 --drop_path 0 --layer_scale_init_value 0.1
> 
> Any idea why this error occurs? Thanks in advance.

Did the pretraining eventually work for you? Thanks!
